File: user-service/src/main/java/com/security/userservice/service/impl/UserServiceImpl.java
package com.security.userservice.service.impl;

import com.security.userservice.domain.Account;
import com.security.userservice.repository.UserRepository;
import com.security.userservice.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Service
@RequiredArgsConstructor
public class UserServiceImpl implements UserService {

    private final UserRepository userRepository;

    @Transactional
    @Override
    public void createAccount(Account account) {

        userRepository.save(account);

    }
}

File: service/src/main/kotlin/com/example/service/service/impl/UserServiceImpl.kt
package com.example.service.service.impl

import com.example.service.data.dto.UserDTO
import com.example.service.data.mapper.UserMapper
import com.example.service.repository.UserRepository
import com.example.service.service.UserService
import mu.KotlinLogging
import org.springframework.stereotype.Service

@Service
class UserServiceImpl(
    private val userMapper: UserMapper, // 정상 동작을 하는 지 체크해 볼 필요가 있다.
    private val userRepository: UserRepository,

): UserService {

    private val logger = KotlinLogging.logger {}

    override fun createUsers(userDTO: UserDTO) {
        logger.info("### .createUsers start")
        userRepository.save(userMapper.toEntity(userDTO));
    }

    override fun deleteUsers(userId: String) {
        logger.info("### .deleteUsers start")

    }

    override fun getUser(userId: String) {
        
    }
}